src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -222,6 +244,12 @@ protected override GetHeadersPayload BuildGetHeadersPayload()
         /// <param name="headers">The headers.</param>
         protected async Task ProcessLegacyHeadersAsync(INetworkPeer peer, List<BlockHeader> headers)
         {
+            if (this.isGateway && !peer.IsWhitelisted())
```

Comment:
Same should be for proven headers payload.

Reply:
I suggest here the same handle gateway and whitelist and remove whitelist form bellow